lib/storage/index_db_legacy.go

```diff
@@ -29,3 +31,45 @@ func mustOpenLegacyIndexDBReadOnly(path string, s *Storage) *indexDB {
 	alwaysReadOnly.Store(true)
 	return mustOpenIndexDB(id, tr, name, path, s, &alwaysReadOnly)
 }
+
+// appendTo scans through all index db items and appends them to the given dstIdbs:
+//   - global index items go to all dstIdbs.
+//   - per-day - only to dstIdbs covering this particular day.
+func (db *indexDB) appendTo(dstIdbs []*indexDB) error {
```

Comment:
I don't like the name, maybe you have a better idea? I also tried `convert`/`migrate`/`transfer`, but they all do not look great.